lib/buildPackage.nix

```diff
@@ -42,7 +42,7 @@ mkCargoDerivation (cleanedArgs // memoizedArgs // {
         installCargoArtifactsMode = args.installCargoArtifactsMode or "use-zstd";
       };
     in
-    buildDepsOnly (removeAttrs depsArgs [ "installPhase" "installPhaseCommand" ])
+    buildDepsOnly (removeAttrs depsArgs [ "installPhase" "installPhaseCommand" "preInstall" "postInstall" ])
```

Comment:
Perhaps instead of trying to clean these up we should just set `installPhase = "mkdir -p $out"` in `depsArgs` which will _not_ run any installation hooks and solve the issue a bit more simply!